services/custom-example/src/giano-injection.ts

```diff
@@ -76,39 +85,44 @@ export function createGianoInjection(storage?: GianoStorage): GianoProviderInjec
       await gianoStorage.setPublicKey(rawId, xyVector);
     },
 
-    // onUserOperationSigned: async (signedUserOp) => {
-    //   try {
-    //     // Submit to backend for validation and bundler submission
-    //     const response = await fetch('/api/submit-userop', {
-    //       method: 'POST',
-    //       body: serializeWithBigInt(signedUserOp),
-    //       headers: { 'Content-Type': 'application/json' },
-    //     });
-
-    //     if (!response.ok) {
-    //       const errorData = await response.json();
-    //       throw new Error(`Backend submission failed: ${errorData.error}`);
-    //     }
-
-    //     const result = await response.json();
-    //     console.log('Backend submission successful:', result);
-
-    //     // Return the transaction receipt from backend
-    //     return result.receipt;
-    //   } catch (error) {
-    //     console.error('UserOp submission failed:', error);
-    //     throw error;
-    //   }
-    // },
+    ...(enableBackendSubmission && {
+      submitUserOperation: async (signedUserOp) => {
+        try {
+          // Submit to backend for validation and bundler submission
+          const response = await fetch('/api/submit-userop', {
+            method: 'POST',
+            body: serializeWithBigInt(signedUserOp),
+            headers: { 'Content-Type': 'application/json' },
+          });
+
+          if (!response.ok) {
+            const errorData = await response.json();
+            throw new Error(`Backend submission failed: ${errorData.error}`);
+          }
+
+          const result = await response.json();
+          console.log('Backend submission successful:', result);
+
+          // Return the transaction receipt from backend
+          return result.receipt;
```

Comment:
The backend should return the hash, so that the frontend can wait for the receipt when it wants to